src/pymor/algorithms/newton.py

```diff
@@ -4,14 +4,16 @@
 
 import numpy as np
 
+from pymor.algorithms.line_search import armijo
+
 from pymor.core.defaults import defaults
 from pymor.core.exceptions import InversionError, NewtonError
 from pymor.core.logger import getLogger
 
 
 @defaults('miniter', 'maxiter', 'rtol', 'atol', 'relax', 'stagnation_window', 'stagnation_threshold')
 def newton(operator, rhs, initial_guess=None, mu=None, error_norm=None, least_squares=False,
-           miniter=0, maxiter=100, rtol=-1., atol=-1., relax=1.,
+           miniter=0, maxiter=100, rtol=-1., atol=-1., relax=1., line_search_method='armijo',
            stagnation_window=3, stagnation_threshold=np.inf,
            return_stages=False, return_residuals=False):
```

Comment:
I think we should add a `line_search_params=None` parameter which can be used to pass a dict of kwargs for the `armijo` method.